.github/workflows/check-state-compatibility.yaml

```diff
@@ -0,0 +1,108 @@
+# This workflow runs on every new commit to the latest development branch. 
+# It checks that the new commit didn't break state compatibility by:
+
+# - building the new `osmosisd` binary with the latest changes
+# - replaying a configurable number of previous blocks from chain history
+
+# Currently, the node starts from a snapshot taken some blocks before the last epoch 
+# and waits `DELTA_HALT_HEIGHT` blocks after epoch before finally halting.
+
+# Important Caveat:
+
+# The fact that this workflow succeeds and the binary doesn't fail doesn't 
+# directly imply that the new binary is state-compatible.
+# It could be that the binary is not state-compatible, but the condition 
+# which would break state compatibility was not present in the chunk of block history used.
+
+# On the other hand, if the workflow fails, the binary is not state-compatible.
+
+name: check_state_compatibility
+
+# ************************************ NOTE ************************************
+# 
+# DO NOT TRIGGER THIS WORKFLOW ON PUBLIC FORKS
+#
+# This workflow runs on a self-hosted runner and forks to this repository 
+# can potentially run dangerous code on the self-hosted runner machine 
+# by creating a pull request that executes the code in a workflow.
+# 
+# ******************************************************************************
+
+on:
+  push:
+    branches:
+      - 'v10.x'
+  pull_request:
+    branches:
+      - 'v10.x'
+  workflow_dispatch:
+    branches:
+      - 'v10.x'
+
+env:
+  SNAPSHOT_URL: https://osmosis-snapshot.sfo3.cdn.digitaloceanspaces.com/osmosis.json
+  RPC_ENDPOINT: https://rpc.osmosis.zone
```

Comment:
We should add fallback RPCs, that way when we get a invalid response we can simple query the next one down the line. If that isn't feasible, we should add some kind of validation to the response and continue querying for x minutes